newrelic/resource_newrelic_workload.go

```diff
@@ -0,0 +1,327 @@
+package newrelic
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/newrelic/newrelic-client-go/pkg/errors"
+	"github.com/newrelic/newrelic-client-go/pkg/workloads"
+)
+
+func resourceNewRelicWorkload() *schema.Resource {
+	userReference := &schema.Schema{
+		Type:     schema.TypeList,
+		MaxItems: 1,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"email": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"gravatar": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"id": {
+					Type:     schema.TypeInt,
+					Computed: true,
+				},
+				"name": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+			},
+		},
+	}
+
+	return &schema.Resource{
+		Create: resourceNewRelicWorkloadCreate,
+		Read:   resourceNewRelicWorkloadRead,
+		Update: resourceNewRelicWorkloadUpdate,
+		Delete: resourceNewRelicWorkloadDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
```

Comment:
In terms of maintainability of the schema, you should try order the attributes that are Required at the top, Optional in the middle, and Computed at the bottom. This makes it fairly easy to scan the schema and know how things are prioritized down when creating the various structs/setting the values in state.

Reply:
👍